### PR TITLE
refactor: move all alloy-compat code to a standalone module

### PR DESCRIPTION
**Pull request by guha-rahul**
addresses #8168 
added the alloy-compact impl from block.rs and header.rs to a new module `compat`

### CODE REVIEW

File: crates/primitives/src/compat/mod.rs
Comment (mattsse) [inline crates/primitives/src/compat/mod.rs:1]: let's name this just alloy_compat.rs

File: crates/primitives/src/lib.rs [hunk 1]
```diff
@@ -59,6 +59,9 @@ pub use chain::{
     ChainSpecBuilder, DisplayHardforks, ForkBaseFeeParams, ForkCondition, NamedChain, DEV, GOERLI,
     HOLESKY, MAINNET, SEPOLIA,
 };
+#[allow(unused_imports)]
+#[cfg(feature = "alloy-compat")]
+use compat::*;
```
Comment (mattsse) [inline crates/primitives/src/lib.rs:64]: this we don't need

File: crates/primitives/src/lib.rs [hunk 2]
```diff
@@ -22,6 +22,7 @@ mod account;
 pub mod basefee;
 mod block;
 mod chain;
+mod compat;
```
Comment (mattsse) [inline crates/primitives/src/lib.rs:25]: this needs a cfg(feature

**Review by mattsse — CHANGES_REQUESTED** [review]
ty,
pedantic nits

**Comment by guha-rahul** [comment]
@mattsse any updates?

**Review by mattsse — APPROVED** [review]
nice, ty

only changed the nested mod

**Pull request merged**